WinCompatibilityPack/WinCompatibilityPack.psm1

```diff
@@ -0,0 +1,804 @@
+
+###########################################################################################
+<#
+.Synopsis
+   Initialize the connection to the compatibility session.
+.DESCRIPTION
+   Initialize the connection to the compatibility session. By default
+   the compatibility session will be created on the local host using the
+   'Microsoft.PowerShell' configuration. On subsequent calls, if a session
+   matching the current specification is found, it will be returned rather than
+   creating a new session. If a matching session is found, but can't be used, it
+   will be closed and a new session will be retrieved.
+
+   This command is called by the other commands in this module so
+   you will rarely call this command directly. 
+.EXAMPLE
+    Initialize-WinSession
+    Initialize the default compatibility session
+.EXAMPLE
+    Initialize-WinSession -ComputerName localhost -ConfigurationName Microsoft.PowerShell
+    Initialize the compatibility session with a specific computer name and configuration
+#>
+function Initialize-WinSession
+{
+    [CmdletBinding()]
+    [Alias("iwins")]
+    [OutputType("System.Management.Automation.Runspaces.PSSession")]
+    Param (
+
+        # If you don't want to use the default compatibility session, use
+        # this parameter to specify the name of the computer on which to create
+        # the compatibility session.
+        # (Defaults to 'localhost')
+        [Parameter(Mandatory=$false,Position=0)]
+        [String]
+        [Alias("Cn")]
+            $ComputerName = $script:DefaultComputerName,
+
+        # Specifies the configuration to connect to when creating the compatibility session
+        # (Defaults to 'Microsoft.PowerShell')
+        [Parameter()]
+        [String]
+            $ConfigurationName = $script:DefaultConfigurationName,
+
+        # The credential to use when connecting to the target machine/configuration
+        [Parameter()]
+        [PSCredential]
+            $Credential,
+
+        # If present, the specified session object will be returned
+        [Parameter()]
+        [Switch]
+            $PassThru,
+
+        # Instead of initializing the session, show what actions will be taken instead.
+        [Parameter()]
+        [Switch]
+            $WhatIf
+    )
+
+    [bool] $verboseFlag = $PSBoundParameters['Verbose']
+
+    if ($ComputerName -eq ".")
+    {
+        $ComputerName = "localhost"
+    }
+
+    Write-Verbose -Verbose:$verboseFlag "Initializing the compatibility session on host '$ComputerName'."
+    if ($Credential)
+    {
+        $script:SessionName = "win-$($Credential.UserName)"
+    }
+    else
+    {
+        $script:SessionName = "win-$((Get-ChildItem env:UserName).Value)"
+    }
+    Write-Verbose -Verbose:$verboseFlag "The compatibility session name is '$script:SessionName'."
+
+    # BUGBUG - need to deal with the possibilities of multiple sessions
+    $session = Get-PSSession | Where-Object {
+        $_.ComputerName      -eq $ComputerName -and
```

Comment:
nit: if you align the comparison operators, I believe it looks better to also align the logic operators:

```powershell
    $session = Get-PSSession | Where-Object {
        $_.ComputerName      -eq $ComputerName       -and
        $_.ConfigurationName -eq $ConfigurationName  -and
        $_.Name              -eq $script:SessionName
    }
```

Reply:
Fixed.